### ISSUE TITLE
Border stroke thickness isn't correct/offset on WinUI

### ISSUE THREAD

**Issue by Foda** [dotnet/maui#16507]
### Description

When setting the `Border.StrokeThickness` in WinUI to 1, the rendered border isn't actually 1px thick. The same happens when set to 3, 5, etc. It seems like the border is inset by half a pixel, so odd values cause it to display wrong :(

![image](https://github.com/dotnet/maui/assets/890772/d49cdbbd-9f2e-46b6-8085-016d1166a930)

### Steps to Reproduce

```xml
<Border
    WidthRequest="300"
    HeightRequest="300"
    StrokeThickness="1"
    Stroke="Red"
    Background="Blue">
        <Label
            Text="Welcome to .NET Multi-platform App UI"
            FontSize="18"
            HorizontalOptions="Center" />
</Border>
```

### Link to public reproduction project repository

N/A

### Version with bug

8.0.0-preview.6.8686

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

All

### Did you find any workaround?

No

**Comment by spadapet** [comment]
@Foda, I built the main branch and the WinUI border thickness is wrong in a different way:

![image](https://github.com/dotnet/maui/assets/2523431/1fe770aa-bbb7-4a7e-abf7-e46b7471eb0f)
* Thickness is half what it should be
* Right and bottom are cut off
* Contents of border are clipped (see top of robot image)
![image](https://github.com/dotnet/maui/assets/2523431/55d4287b-7741-47c0-90d1-4fb7890b0b67)

Make the thickness super large to see the right/bottom:

![image](https://github.com/dotnet/maui/assets/2523431/a75a8f7f-7477-4ef1-a8d9-a11e2c86b882)
![image](https://github.com/dotnet/maui/assets/2523431/1ed63158-3592-47c7-b428-0a12a1e50cec)

If I remove the margin, the border is way out of bounds:

![image](https://github.com/dotnet/maui/assets/2523431/ce9287c8-9cd4-42e7-9047-9d175727a811)
![image](https://github.com/dotnet/maui/assets/2523431/3be993c7-b2ea-4857-9e30-f60fde8bf38a)

**Comment by Foda** [comment]
Yeah, it’s…. totally messed up. I also wonder if it’d be better to use the WinUI border if we know the shape is a rect or rounded rect, but we should figure out what’s causing this first
________________________________

**Comment by cgp1976** [comment]
I am facing this issues as well on Visual Studio 2022 17.8 Preview 1 8.0.0-preview.7.8842. literally just the top and left border are rendered.

**Comment by XamlTest** [comment]
Verified this on Visual Studio Enterprise 17.8.0 Preview 1.0. Repro on Windows 11, Android 13.0-API33 and iOS 16.4 .NET 8 with below Project:
[16507.zip](https://github.com/dotnet/maui/files/12357745/16507.zip)

**Comment by afernandes** [comment]
Border.Stroke and Border.StrokeThickness incorrect.

the border color is lighter than the defined color and the border thickness is too large even though it is set to 1.
I have added a frame for comparison:

![image](https://github.com/user-attachments/assets/f318e387-eca5-4d68-beed-fea6873105cf)
``` xaml
<Border
    Stroke="Black"
    StrokeThickness="1"
    StrokeShape="RoundRectangle 8"
    Background="White"
    Padding="16,8"
    HorizontalOptions="Center">
    <Label
        Text=".NET MAUI"
        TextColor="Black"
        FontSize="18"
        FontAttributes="Bold" />
</Border>

<Frame
    BorderColor="Black"
    CornerRadius="8"
    BackgroundColor="White"
    Padding="16,8"
    HorizontalOptions="Center">
    <Label
        Text=".NET MAUI"
        TextColor="Black"
        FontSize="18"
        FontAttributes="Bold" />
</Frame>
```
